doc/source/whatsnew/v0.23.0.txt

```diff
@@ -119,6 +119,61 @@ Current Behavior
 
     s.rank(na_option='top')
 
+
+.. _whatsnew_0230.enhancements.assign_dependent:
+
+``.assign()`` accepts dependent arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :func:`DataFrame.assign()` now accepts dependent kwargs. (:issue: `14207)
+
+Specifically, defining a new column inside ``.assign()`` may be referenced in the same assign statement if a callable is used. For example
+
+.. ipython:: python     
+
+    In [3]: df = pd.DataFrame({'A': [1, 2, 3]})
+
+    In [4]: df.assign(B=df.A, C=lambda x:x['A']+ x['B'])
+    Out[4]:
+       A  B  C
+    0  1  1  2
+    1  2  2  4
+    2  3  3  6
+
+.. warning::
+
+This may subtly change the behavior of your code when you're
+using ``.assign()`` to update an existing column. Previously, callables
+referring to other variables being updated would get the "old" values
+
+Previous Behaviour:
+
+.. code-block:: ipython
+
+    In [2]: df = pd.DataFrame({"A": [1, 2, 3]})
+
+    In [3]: df.assign(A=lambda df: df.A + 1, C=lambda df: df.A * -1)
+    Out[3]:
+       A  C
+    0  2 -1
+    1  3 -2
+    2  4 -3
+
+New Behaviour:
+
+.. ipython:: python
```

Comment:
but you don't need all this, just

```
df.assign(....) # complete the line of course
```

generally we would simply execute and show the DataFrame creation in an ipython block at the top of the sub-section, then do a code-block for previous, and an ipython block for enw

Reply:
see above